include/mbgl/style/layer.hpp

```diff
@@ -117,14 +111,12 @@ class Layer : public mbgl::util::noncopyable {
     void setMaxZoom(float) const;
 
     // Private implementation
-    const std::unique_ptr<Impl> baseImpl;
+    const std::shared_ptr<Impl> baseImpl;
```

Comment:
@jfirebaugh If we copy layers in `GeometryTile` to pass to `GeometryTileWorker::setLayers` and then create `RenderLayer`s from that in the `GeometryTileWorker`; we would be copying the associated `Impl` twice right? In that case, maybe we should revert the changes to `GeometryTileWorker` and group_by_layout.hpp to continue to work on `Layer`s?